Improve accessibility

AylinUenal opened the pull request:
## Description

We adjusted the swatches components to improve accessibility for screen readers. It is now possible to select color and size of a product variant.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

Commit:
CURB-4816 make swatches selectable by screen reader, bump version 1.5.0

Commit:
CURB-4816 make swatches accessible in product list page

SaschaKrist commented:
Just some thoughts. When "Color" has labels the labels will be rendered. Seems like the the swatchLabels.enabled is global. Maybe it should be per variant.

Commit:
CURB-4816 improve a11y in swatches

Commit:
CURB-4816 adjust aria-labels of foldable swatches

Commit:
CURB-4816 fix aria-label of foldable swatches

fkloes commented on frontend/components/PdpColorSwatch/index.jsx:
```diff
@@ -52,6 +52,7 @@ const PdpColorSwatch = ({
       swatchColor: value.color,
       selected: characteristics && characteristics[swatch.id] === value.id,
       selectable: true,
+      showSwatchLabel: swatchLabels.labels.Color,
```
you assign strings (and sometimes false) to a variable that sounds at it was a boolean.

when the value is false, swatches are now hidden. what was the reason to introduce new behaviour to the extension? the code that used `swatchLabels.labels` till now creates a variable called `label` which is not used in the components that now render empty when the value is false.

AylinUenal replied:
I came up with this solution because for the correct color in the aria label, the merchant needs to define the swatchlabel but they should not show up, otherwise it looks like this:
<img width="264" alt="Screenshot 2025-05-06 at 08 55 16" src="https://github.com/user-attachments/assets/369b087b-2723-4dd7-a3ec-f643cc729f9f" />
This is not to be confused with the label itself which shows above the swatches, e.g. the label "Farbe".

fkloes commented on frontend/components/PdpSizeSwatch/index.jsx:
```diff
@@ -55,6 +55,7 @@ const PdpSizeSwatch = ({
     return swatch.values.map(value => ({
       ...value,
       swatchLabel: value.label,
+      showSwatchLabel: swatchLabels.labels.Size,
```
same feedback as for `frontend/components/PdpColorSwatch/index.jsx`

fkloes commented on frontend/components/PlpSwatches/index.jsx:
```diff
@@ -14,9 +15,13 @@ const PlpSwatches = ({ swatches }) => {
     return null;
   }
 
+  const ariaLabel = i18n.text('variants', {
```
please wrap this in a `useMemo`, so that i doesn't have to be recreated whenever the component renders.

AylinUenal replied:
done

fkloes commented on frontend/components/Swatch/index.jsx:
```diff
@@ -26,21 +26,27 @@ const styles = {
     )}
+    role="option"
+    tabIndex={0}
+    aria-selected={ariaSelected}
```
why don't you add the label to the `Tag`? For color swatches the aria label is a hex code. Not sure if a lot of humans know which color it represents.

AylinUenal replied:
the hex code is only the fallback value in case the swatchlabel is not there

Commit:
CURB-4816 small fixes after review

fkloes commented on frontend/components/PlpSwatches/index.jsx:
```diff
@@ -10,13 +11,19 @@ import { maxSwatches } from '../../config';
+    i18n.text('variants', {
+      swatches: swatches.length,
+    });
+  }, [swatches.length]);
```
seems that the swatches prop is not always an array

AylinUenal replied:
fixed

Commit:
CURB-4816 revert swatch label check

fkloes committed:
CURB-4816 Simplified structure of Swatch component

- replaced Link component by internal logic inside Swatch component
- allowed to implement more reliable a11y annotations

fkloes committed:
CURB-4816 Improved a11y for swatch list label
